NorthstarDLL/logging/logging.cpp

```diff
@@ -220,3 +222,94 @@ void NS::log::FlushLoggers()
 
 	spdlog::default_logger()->flush();
 }
+
+// Wine specific functions
+typedef const char *(CDECL *wine_get_host_version_type)(const char**, const char**);
+wine_get_host_version_type wine_get_host_version;
+
+typedef const char *(CDECL *wine_get_build_id_type)(void);
+wine_get_build_id_type wine_get_build_id;
+
+typedef char *(CDECL *wine_get_unix_file_name_type)(LPCWSTR);
+wine_get_unix_file_name_type wine_get_unix_file_name;
+
+typedef WCHAR *(CDECL *wine_get_dos_file_name_type)(LPCSTR);
+wine_get_dos_file_name_type wine_get_dos_file_name;
+
+void StartupLog()
+{
+	spdlog::info("NorthstarLauncher version: {}", version);
+	spdlog::info("Command line: {}", GetCommandLineA());
+	spdlog::info("Using profile: {}", GetNorthstarPrefix());
+
+	HMODULE ntdll = GetModuleHandleA("ntdll.dll");
+	if(!ntdll)
+	{
+		// How did we get here
+		spdlog::info("Operating System: Unknown");
+		return;
+	}
+
+	wine_get_host_version = (wine_get_host_version_type)GetProcAddress(ntdll, "wine_get_host_version");
+	if (wine_get_host_version)
+	{
+		// Load the rest of the functions we need
+		wine_get_build_id = (wine_get_build_id_type)GetProcAddress(ntdll, "wine_get_build_id");
+    	wine_get_unix_file_name = (wine_get_unix_file_name_type)GetProcAddress(GetModuleHandleA("KERNEL32"), "wine_get_unix_file_name");
+    	wine_get_dos_file_name = (wine_get_dos_file_name_type)GetProcAddress(GetModuleHandleA("KERNEL32"), "wine_get_dos_file_name");
+
+		const char* sysname;
+		wine_get_host_version(&sysname, NULL);
+
+		spdlog::info("Operating System: {} (Wine)", sysname);
+		spdlog::info("Wine build: {}", wine_get_build_id());
+
+		char* compatToolPtr = std::getenv("STEAM_COMPAT_TOOL_PATHS");
+		if (compatToolPtr)
+		{
+			std::string compatToolPath(compatToolPtr);
+
+			spdlog::info("Proton build: {}", compatToolPath.substr(compatToolPath.rfind("/") + 1));
+		}
+
+
+	}
+	else
+	{
+		// We are real Windows (hopefully)
+		const char* win_ver = "Unknown";
+
+    	OSVERSIONINFO osvi;
+    	BOOL bIsWindowsXPorLater;
+
+    	ZeroMemory(&osvi, sizeof(OSVERSIONINFO));
+	    osvi.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+
+		GetVersionEx(&osvi);
+
+		// Anything prior to 6 is Vista, and I don't think we need to support those
+		if (osvi.dwMajorVersion == 6)
+		{
+			switch (osvi.dwMinorVersion)
+			{
+				case 1:
+					win_ver = "7";
+					break;
+				case 2:
+					win_ver = "8";
+					break;
+				case 3:
+					win_ver = "8.1";
+					break;
+				default:
+					break;
+			}
+		}
+		else if (osvi.dwMajorVersion == 10)
+		{
+			win_ver = "10";
+		}
```

Comment:
The relevant table I was consulting did not have an entry for Windows 11.

I assume its just `dwMajorVersion == 11`, but I have no setup to test this (neither do I really want to install Windows 11)